File: scripts/dataset_lib/tpch.py
import re

from dataset_lib.common import *


def tpch_get_table_list():
    return tpch.table_list


class tpch:
    table_list = \
        [
            'customer',  #
            'lineitem',  #
            'nation',
            'orders',  #
            'part',
            'partsupp',
            'region',
            'supplier'
            #
        ]
    name = "tpch"

    def __init__(self):
        pass

File: test/src/dataset_tests/tpch.cpp
#include "data/tpch.hpp" // Make sure this header defines Tpch::customer, Tpch::lineitem, etc.
#include "fastlanes.hpp"
#include "fls_tester.hpp"

namespace fastlanes {

// Test reading/processing for each TPCH table. Uses TestCorrectness to compare
// against a reference CSV or expected output. Adjust the test names as needed.

TEST_F(FastLanesReaderTester, tpch_customer) {
	TestCorrectness(Tpch::customer);
}

TEST_F(FastLanesReaderTester, tpch_lineitem) {
	TestCorrectness(Tpch::lineitem);
}

TEST_F(FastLanesReaderTester, tpch_orders) {
	TestCorrectness(Tpch::orders);
}

TEST_F(FastLanesReaderTester, tpch_part) {
	TestCorrectness(Tpch::part);
}

TEST_F(FastLanesReaderTester, tpch_partsupp) {
	TestCorrectness(Tpch::partsupp);
}

} // namespace fastlanes
